### DISCUSSION TITLE
Interacting with value in another component

### DISCUSSION THREAD

**Discussion by jochemgruter** [alpinejs/alpine#914] [1. Help]
I am building a blade toggle component using AlpineJS. The idea is that you can enable/disable the UI toggle and a invissible checkbox will be put on/off

toggle.blade.php: 

    <div x-data="{on: false}">
        <button type="button" @click="on = !on" :class="{'bg-cyan-600' : on, 'bg-gray-200' : !on}">
            <span :class="{'translate-x-5' : on, 'translate-x-0' : !on}" class="..."></span>
        </button>

        <input type="checkbox" name="{{$name}}" x-model="on" value="on" class="hidden" />
    </div>

This is working good so far. I can put the toggle in my form and toggle it:

    <x-toggle name="test" />

But now I want other elements interact with this toggle. For example make another input field vissible by toggling the test-toggle on. 

So it need to work something like this:

    <div x-data="{show: false}">
        <x-toggle name="test" x-model="show" />
        
        <div x-show="show">
            
        </div>
    </div>

How can I achive this?

**Comment by KevinBatdorf** [comment]
`click` events bubble up so you could add a `click` listener on the outer div and it will work. 

```html
<div x-data="{ show: false }" @click="show = true">
```

Otherwise if you need something more complex you could `$dispatch()` inside a `$watch()`

Like this: https://codepen.io/KevinBatdorf/pen/mdrVwOX

**Reply by jochemgruter** [reply]
The $dispatch function works great. Thank you.
